From: William Wold <wm@wmww.sh>
Date: Sun, 5 Apr 2020 05:26:24 -0400
Subject: Do not require an initial configure for custom Wayland surfaces

There is no way for custom Wayland surfaces to get configure events, so an
initial configure event should not be required to resize a custom surface.

Bug: https://gitlab.gnome.org/GNOME/gtk/-/issues/2578
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=955820
Forwarded: https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/1633
Applied-upstream: 3.24.18, commit:12fc9a45efcbb546eb7de13c5c4d3183f2f5a3b8
---
 gdk/wayland/gdkwindow-wayland.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gdk/wayland/gdkwindow-wayland.c b/gdk/wayland/gdkwindow-wayland.c
index eba8361..5f39c57 100644
--- a/gdk/wayland/gdkwindow-wayland.c
+++ b/gdk/wayland/gdkwindow-wayland.c
@@ -1087,6 +1087,8 @@ needs_initial_configure (GdkWindow *window)
 
   if (impl->display_server.wl_subsurface)
     return FALSE;
+  else if (impl->use_custom_surface)
+    return FALSE;
   else if (is_realized_toplevel (window))
     return TRUE;
   else if (is_realized_popup (window))
